WcaOnRails/app/controllers/admin_controller.rb

```diff
@@ -291,43 +291,57 @@ def edit_person
   end
 
   def update_person
-    @person = Person.current.find_by(wca_id: params[:person][:wca_id])
-    if @person
-      person_params = params.require(:person).permit(:name, :countryId, :gender, :dob, :incorrect_wca_id_claim_count)
-      case params[:method]
-      when "fix"
-        if @person.update(person_params)
-          flash.now[:success] = "Successfully fixed #{@person.name}."
-          if @person.saved_change_to_countryId?
-            flash.now[:warning] = "The change you made may have affected national and continental records, be sure to run
-            <a href='#{admin_check_regional_records_path}'>check_regional_record_markers</a>.".html_safe
-          end
-        else
-          flash.now[:danger] = "Error while fixing #{@person.name}."
-        end
-      when "update"
-        if @person.update_using_sub_id(person_params)
-          flash.now[:success] = "Successfully updated #{@person.name}."
-        else
-          flash.now[:danger] = "Error while updating #{@person.name}."
-        end
-      when "destroy"
-        if @person.results.any?
-          flash.now[:danger] = "#{@person.name} has results, can't destroy them."
-        elsif @person.user.present?
-          flash.now[:danger] = "#{@person.wca_id} is linked to a user, can't destroy them."
-        else
-          name = @person.name
-          @person.destroy
-          flash.now[:success] = "Successfully destroyed #{name}."
-          @person = Person.new
-        end
+    person_params = params.require(:person)
+    wca_id = person_params.require(:wcaId)
+    person = Person.current.find_by(wca_id: wca_id)
+
+    if person.nil?
+      render status: :unprocessable_entity, json: { error: "Person with WCA ID #{wca_id} not found." }
+      return
+    end
+
+    name = person_params.require(:name)
+    representing = person_params.require(:representing)
+    gender = person_params.require(:gender)
+    dob = person_params.require(:dob)
+    country_id = Country.find_by_iso2(representing).id
+
+    edit_params = {
+      name: name,
+      countryId: country_id,
+      gender: gender,
+      dob: dob,
+      incorrect_wca_id_claim_count: 0,
+    }
+    case params[:method]
```

Comment:
I am not a big fan of having a switch-case for which action to take.
The design paradigm behind REST APIs should be to make a different route (possibly with a different HTTP verb) for different actions. This case is a little bit special because `fix` and `update` are two very similar actions, so I would potentially agree to push them into one common `PATCH` route.
But `"destroy"` should definitely be a separate HTTP `DELETE` route, and `"reset-claim-count"` should be a separate HTTP `PUT` with own, individual endpoints.